File: src/main/java/dev/lampirg/consultationappointment/data/appointment/validator/InTimeBordersValidator.java
package dev.lampirg.consultationappointment.data.appointment.validator;

import dev.lampirg.consultationappointment.data.appointment.Appointment;
import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public class InTimeBordersValidator implements ConstraintValidator<InTimeBorders, Appointment> {
    @Override
    public boolean isValid(Appointment appointment, ConstraintValidatorContext constraintValidatorContext) {
        boolean isAfterStart =
                appointment.getStartTime().isAfter(appointment.getAppointmentPeriod().getStartTime()) ||
                appointment.getStartTime().isEqual(appointment.getAppointmentPeriod().getStartTime());

        boolean isBeforeEnd = appointment.getStartTime().isBefore(
                appointment.getAppointmentPeriod().getEndTime()
        );
        return isAfterStart && isBeforeEnd;
    }
}

File: src/main/kotlin/dev/lampirg/consultationappointment/data/appointment/validator/InTimeBordersValidator.kt
package dev.lampirg.consultationappointment.data.appointment.validator

import dev.lampirg.consultationappointment.data.appointment.Appointment
import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class InTimeBordersValidator : ConstraintValidator<InTimeBorders?, Appointment> {
    override fun isValid(appointment: Appointment, constraintValidatorContext: ConstraintValidatorContext): Boolean {
        val isAfterStart = appointment.startTime.isAfter(appointment.appointmentPeriod.startTime) ||
                appointment.startTime.isEqual(appointment.appointmentPeriod.startTime)
        val isBeforeEnd: Boolean = appointment.startTime.isBefore(appointment.appointmentPeriod.endTime)
        return isAfterStart && isBeforeEnd
    }
}